pkg/eventpersister/persister/persister.go

```diff
@@ -688,3 +688,43 @@ func (p *Persister) getUserEvaluation(
 	}
 	return evaluation, nil
 }
+
+func (p *Persister) setExpiration(dCmd *goredis.DurationCmd, key string) error {
+	// The value of command is available only after the pipeline is executed.
+	// https://redis.uptrace.dev/guide/go-redis-pipelines.html#pipelines
+	d, err := dCmd.Result()
+	if err != nil {
+		return err
+	}
+	if d == -1 {
+		// The expiration may be overriden because of race condition.
+		// However, we don't have to care it because this expiration is not have to be strict.
+		_, err := p.evaluationCountCacher.Expire(key, oneMonth)
```

Comment:
Even though this implementation makes sure that expiration is set, we have to allow the following case:

1. Persister failed to set expiration.
2. The next evaluation event is sent after 15 days.
3. The first setting evaluation counts will be removed after 46 days, not 31 days.

Reply:
Okimoto-san said it's a rare case, so we don't have to care it.